caddyhttp/markdown/setup.go

```diff
@@ -41,6 +41,11 @@ func setup(c *caddy.Controller) error {
 
 	cfg := httpserver.GetConfig(c)
 
+	// Add markdown index files to *SiteConfig.IndexPages
+	for _, mdc := range mdconfigs {
+		cfg.IndexPages = append(cfg.IndexPages, mdc.IndexFiles...)
```

Comment:
If someone could help to de-duplicate the resulting string, that would be a great help.  

With this implementation, the resulting IndexPages can contain many duplicates like this: [index.html index.htm index.txt default.html default.htm default.txt index.html index.txt index.md index.txt index.md index.html index.json]

Reply:
Or just iterate the list (it's usually small, less than ~30 items) and see if it exists, and if not, then add it.

Reply:
@Mohammed90 Unfortunately, converting it to a map makes it lose the order of files, which in this case is important.